internal/metrics/querynode.go

```diff
@@ -0,0 +1,273 @@
+			"num_segments",
+		})
+
+	QueryNodeNumDmlChannels = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_dml_channels",
+			Help:      "Number of dmlChannels per collection in QueryNode.",
+		}, []string{
+			"num_dml_channels",
+		})
+
+	QueryNodeNumDeltaChannels = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_delta_channels",
+			Help:      "Number of deltaChannels per collection in QueryNode.",
+		}, []string{
+			"num_delta_channels",
+		})
+
+	QueryNodeNumConsumers = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_consumers",
+			Help:      "Number of consumers per collection in QueryNode.",
+		}, []string{
+			"num_consumers",
+		})
+
+	QueryNodeNumReaders = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_readers",
+			Help:      "Number of readers per collection in QueryNode.",
+		}, []string{
+			"num_readers",
+		})
+
+	QueryNodeSearchCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "search_count",
+			Help:      "Search_count request statistic in QueryNode.",
+		}, []string{
+			"status",
+		})
+
+	QueryNodeRetrieveCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "retrieve_count",
+			Help:      "Retrieve request statistic in QueryNode.",
+		}, []string{
+			"status",
+		})
+
+	QueryNodeQueryDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "query_duration",
+			Help:      "The query durations in QueryNode, query means search or retrieve.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"query_type",
+		})
+
+	QueryNodeQueryDurationInUnsolvedBuffer = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "query_duration_in_unsolved",
+			Help:      "The query durations in unsolved buffer in QueryNode, query means search or retrieve.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"query_type",
+		})
+
+	QueryNodeQueryDurationOnSealedSegments = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "query_duration_on_sealed",
+			Help:      "The query durations on sealed segments, query means search or retrieve.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"query_type",
+		})
+
+	QueryNodeQueryDurationOnGrowingSegments = prometheus.NewHistogramVec(
```

Comment:
rename to QuerNodeSQSegmentLantench
use label to distinguish search and query

Reply:
fixed